src/main/java/io/cdap/plugin/http/common/http/OAuthUtil.java

```diff
@@ -70,12 +77,76 @@ public static AccessToken getAccessToken(BaseHttpConfig config) throws IOExcepti
         return OAuthUtil.getAccessTokenByServiceAccount(config);
       case OAUTH2:
         try (CloseableHttpClient client = HttpClients.createDefault()) {
-          return OAuthUtil.getAccessTokenByRefreshToken(client, config);
+          return getAccessToken(client, (BaseHttpSourceConfig) config);
         }
     }
     return null;
   }
 
+  public static AccessToken getAccessToken(CloseableHttpClient httpclient,
+      BaseHttpSourceConfig config) throws IOException {
+    switch (config.getOauth2GrantType()) {
+      case REFRESH_TOKEN:
+        return getAccessTokenByRefreshToken(httpclient, config);
+      case CLIENT_CREDENTIALS:
+        return getAccessTokenByClientCredentials(httpclient, config.getTokenUrl(),
+            config.getClientId(), config.getClientSecret(), config.getScopes(),
+            config.getOauth2ClientAuthentication().getValue());
+      default:
+        throw new IOException("Invalid Grant Type. Cannot retrieve access token.");
+    }
+  }
+
+  private static AccessToken getAccessTokenByClientCredentials(CloseableHttpClient httpclient,
+      String tokenUrl, String clientId, String clientSecret, String scope,
+      String clientAuthentication) throws IOException {
+    URI uri;
+    try {
+      uri = new URIBuilder(tokenUrl).build();
+    } catch (URISyntaxException e) {
+      throw new IllegalArgumentException(
+          "Failed to build access token URI for OAuth2 with grant type = "
+              + OAuthGrantType.CLIENT_CREDENTIALS.getValue(), e);
+    } catch (NullPointerException e) {
+      throw new IllegalArgumentException(
+          "One or more required OAuth2 parameters (Client ID, Client Secret, "
+              + "or Token URL) are missing.", e);
+    }
+
+    HttpPost httppost = new HttpPost(uri);
+    List<BasicNameValuePair> nameValuePairs = new ArrayList<>();
+    nameValuePairs.add(new BasicNameValuePair("scope", scope));
+    nameValuePairs.add(
+        new BasicNameValuePair("grant_type", OAuthGrantType.CLIENT_CREDENTIALS.getValue()));
+    nameValuePairs.add(new BasicNameValuePair("client_authentication", clientAuthentication));
```

Comment:
we dont need client_authentication header in request, that is required when we are making request and check whether to set params in request or body